mrmustard/lab_dev/transformations/base.py

```diff
@@ -131,98 +150,82 @@ def __rshift__(self, other: CircuitComponent) -> CircuitComponent:
         ret = super().__rshift__(other)
 
         if isinstance(other, Unitary):
-            return Unitary._from_attributes(ret.representation, ret.wires, "")
+            return Unitary._from_attributes(ret.representation, ret.wires)
         elif isinstance(other, Channel):
-            return Channel._from_attributes(ret.representation, ret.wires, "")
+            return Channel._from_attributes(ret.representation, ret.wires)
         return ret
 
-    def __repr__(self) -> str:
-        return super().__repr__().replace("CircuitComponent", "Unitary")
-
     @classmethod
-    def from_bargmann(
+    def from_symplectic(
         cls,
         modes: Sequence[int],
-        triple: tuple[ComplexMatrix, ComplexVector, complex],
+        symplectic: RealMatrix,
+        displacement: RealVector,
         name: Optional[str] = None,
     ) -> Unitary:
-        A = math.astensor(triple[0])
-        b = math.astensor(triple[1])
-        c = math.astensor(triple[2])
-        shape_check(A, b, 2 * len(modes), "Bargmann")
-        s = set(modes)
+        r"""Initialize a Unitary from the given symplectic matrix in qqpp basis.
+        I.e. the axes are ordered as [q0, q1, ..., p0, p1, ...].
+        """
+        if symplectic.shape[-2:] != (2 * len(modes), 2 * len(modes)):
+            raise ValueError(
+                "Symplectic matrix and number of modes don't match. "
+                + f"Modes imply shape {(2 * len(modes), 2 * len(modes))}, "
+                + f"but shape is {symplectic.shape[-2:]}."
```

Comment:
no need to change here, but python pro tip - strings are auto-concatenated:
```pycon
>>> "foo""bar" == "foobar"
True
```
so you can do things like:
```python
x = (
    "my first long line "
    f"and my second longer line with variable {y}"
)
```